[BE] Use ValueError for input.dim check in torch.nn.modules

isdanni opened the pull request:
Summary: Use ValueError for input.dim check instead of Assertion Error.

Fix: #104839

Test Plan: Please see GitHub actions.

Differential Revision: D47427998

albanD reviewed (approved):
Nice improvement, thanks!

Skylion007 reviewed (commented):
Some nit fixes

isdanni committed:
[BE] Use ValueError for input.dim check in torch.nn.modules (#105127)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/105127

Use ValueError for input.dim check instead of Assertion Error.

Test Plan: Please see GitHub actions.

Differential Revision: D47427998

fbshipit-source-id: 0061f6a9de9614bb044c47f2c17902745c01af62

isdanni commented on torch/nn/modules/activation.py:
```diff
@@ -1526,7 +1526,8 @@ class Softmax2d(Module):
 
     def forward(self, input: Tensor) -> Tensor:
-        assert input.dim() == 4 or input.dim() == 3, 'Softmax2d requires a 3D or 4D tensor as input'
+        if input.dim() != 3 and input.dim() != 4:
```
Good suggestion! I updated the diff.

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge
